src/binder_v2/drop.rs

```diff
@@ -0,0 +1,137 @@
+use std::result::Result as RawResult;
+use std::str::FromStr;
+
+use serde::{Deserialize, Serialize};
+
+use super::*;
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Copy, Clone, Serialize, Deserialize)]
+pub struct BindDrop {
+    pub object: Object,
+    pub if_exists: bool,
+    pub cascade: bool,
+}
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Copy, Clone, Serialize, Deserialize)]
+pub enum Object {
+    Table(TableRefId),
+}
+
+impl std::fmt::Display for BindDrop {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "object: {}, exists: {}, cascade: {}",
+            self.object, self.if_exists, self.cascade,
+        )
+    }
+}
+
+impl std::fmt::Display for Object {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Object::Table(table_id_ref) => write!(f, "{}", table_id_ref),
+        }
+    }
+}
+
+#[derive(thiserror::Error, Debug, Clone)]
+#[error("parse drop id error: {}")]
+pub enum ParseDropError {
+    #[error("no leading '$'")]
+    NoLeadingDollar,
+    #[error("invalid table")]
+    InvalidTable,
+    #[error("invalid number: {0}")]
+    InvalidNum(#[from] std::num::ParseIntError),
+    #[error("invalid bool: {0}")]
+    InvalidBool(#[from] std::str::ParseBoolError),
+}
+
+impl FromStr for BindDrop {
+    type Err = ParseDropError;
+
+    fn from_str(s: &str) -> RawResult<Self, Self::Err> {
```

Comment:
`FromStr` is used for parsing an egg expression from a LISP-style S-expression. It's hard to design a format for such a structure and maybe we will never construct one from the string. So I suggest just returning an error in this function like:
```rust
impl FromStr for BindDrop {
    type Err = ();
    fn from_str(s: &str) -> Result<Self, Self::Err> { Err(()) }
}
```